### DISCUSSION TITLE
Weird initial snap when transitioning between PhantomCameras

### DISCUSSION THREAD

**Discussion by deezgamezdev** [ramokz/phantom-camera#436] [Question / Help Wanted] [answered]
Hi, so basically I have this setup on my game.

Player has a main camera and a secondary camera attached (both PhantomCameras). Main camera acts as the name intends to, rotating according to mouse events.

Now for the secondary camera, when we get close to an object that has a camera attached (not PhantomCamera), we grab the transform of that and paste into our secondary camera.

When the player interacts with the computer for example, I just set the priority of the secondary camera to a high number, when there is a second interaction, I set the priority back to 0.

Everything works great except when the transition triggers I get this weird inital snap / tilt to the angle of the object im facing.
Video demonstrates transitioning from several directions.

Any idea why this is happening ?
Thanks in advance.

https://github.com/user-attachments/assets/35748ff1-2e27-4fef-a39e-8676da031218

**Comment by ramokz** [comment]
> we grab the transform of that and paste into our secondary camera.

My guess would be that you're changing the transform of the `PCam3D` right at some point around the tween?

The transform change you mention seems to suggest that the tween's starting position, i.e the `PCam3D`'s position, changes right before the tween kicks in?

The end of the tween also seems to snap, which feels like a similar situation?

**Reply by deezgamezdev** [reply]
That was also my first thought and I was setting the transform on interact previously.

But then I changed it to set the transform when the player looks and detects an interactable that has a camera attached without interacting to it.. the player cam I am only changing the rotation with the mouse and I lock it as soon as i interact.

**Reply by ramokz** [reply]
If you're willing to share a simple sample project / scene, can be without any specific assets, then I'm happy to take a look.

**Reply by deezgamezdev** [reply, marked as answer]
I was building the scene for you, and actually figured out what was going on.

For some reason I left the PhantomCameraHost and the main Camera3D that the host uses in the player scene... and this camera3d was not top level.

Turning that top level switch on the Root Camera fixed it and now the transition is super smooth.

Thanks alot, also because if I wasn't going to build the scene for you maybe I wouldn't figure it out.

Amazing plugin, can't wait for the multi hosts version to come !
